conans/client/command.py

```diff
@@ -1342,6 +1342,8 @@ def link(self, *args):
         parser.add_argument('reference', help='Reference to link. e.g.: mylib/1.X@user/channel')
         parser.add_argument("--remove", action='store_true', default=False,
                             help='Remove linked reference (target not required)')
+        parser.add_argument("-l", "--layout",
+                            help='Define an alternative layout filename')
```

Comment:
IMO, this should be a path to a file, not just a name. Maybe we can agree that relative paths are resolved to the _layouts_ folder inside the cache (here you have the _layout name_), but this should be converted to an absolute path as soon as possible and this path stored associated with the reference. And by default, this argument should take the _default layout_ name/path.

The logic in the installer will be straightforward then.

Reply:
Tested, this arguments allows relative paths to files, both in the edited repo and in the local cache.

Reply:
I think we already had misleading situations when profile name incorrectly resolved to the file with the same name incidentally appeared in the same folder during build, which was hard to debug

Reply:
Shouldn't that case just raise an error saying that the file is incorrect?
I could add tests with random other files to check error messages

Reply:
probably, better to fail early if such coincidence happen, at least it will help to debug

Reply:
Added a test for outputing a clear error when another file is parsed.

Reply:
Need to check what it is stored in the cache, I would want to see there an absolute path (easier for everything I think).